From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 17 Nov 2020 14:49:31 +0100
Subject: core: Move away from clutter_event_get_device_id()

Fetch the device, and the ID from there.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1569>
---
 src/core/events.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/core/events.c b/src/core/events.c
index 438beda..eb679be 100644
--- a/src/core/events.c
+++ b/src/core/events.c
@@ -163,6 +163,7 @@ maybe_unfreeze_pointer_events (MetaBackend          *backend,
                                const ClutterEvent   *event,
                                EventsUnfreezeMethod  unfreeze_method)
 {
+  ClutterInputDevice *device;
   Display *xdisplay;
   int event_mode;
   int device_id;
@@ -173,7 +174,8 @@ maybe_unfreeze_pointer_events (MetaBackend          *backend,
   if (!META_IS_BACKEND_X11 (backend))
     return;
 
-  device_id = clutter_event_get_device_id (event);
+  device = clutter_event_get_device (event);
+  device_id = clutter_input_device_get_device_id (device);
   switch (unfreeze_method)
     {
     case EVENTS_UNFREEZE_SYNC:
